patches/minecraft/net/minecraft/potion/PotionEffect.java.patch

```diff
@@ -41,6 +41,15 @@
          }
      }
  
+@@ -232,7 +236,7 @@
+     public int compareTo(PotionEffect p_compareTo_1_)
+     {
+         int i = 32147;
+-        return (this.func_76459_b() <= 32147 || p_compareTo_1_.func_76459_b() <= 32147) && (!this.func_82720_e() || !p_compareTo_1_.func_82720_e()) ? ComparisonChain.start().compare(Boolean.valueOf(this.func_82720_e()), Boolean.valueOf(p_compareTo_1_.func_82720_e())).compare(this.func_76459_b(), p_compareTo_1_.func_76459_b()).compare(this.func_188419_a().func_76401_j(), p_compareTo_1_.func_188419_a().func_76401_j()).result() : ComparisonChain.start().compare(Boolean.valueOf(this.func_82720_e()), Boolean.valueOf(p_compareTo_1_.func_82720_e())).compare(this.func_188419_a().func_76401_j(), p_compareTo_1_.func_188419_a().func_76401_j()).result();
++        return (this.func_188419_a().getSortIndex() == p_compareTo_1_.func_188419_a().getSortIndex()) ? (this.func_76459_b() <= 32147 || p_compareTo_1_.func_76459_b() <= 32147) && (!this.func_82720_e() || !p_compareTo_1_.func_82720_e()) ? ComparisonChain.start().compare(Boolean.valueOf(this.func_82720_e()), Boolean.valueOf(p_compareTo_1_.func_82720_e())).compare(this.func_76459_b(), p_compareTo_1_.func_76459_b()).compare(this.func_188419_a().func_76401_j(), p_compareTo_1_.func_188419_a().func_76401_j()).result() : ComparisonChain.start().compare(Boolean.valueOf(this.func_82720_e()), Boolean.valueOf(p_compareTo_1_.func_82720_e())).compare(this.func_188419_a().func_76401_j(), p_compareTo_1_.func_188419_a().func_76401_j()).result() : Integer.valueOf(this.func_188419_a().getSortIndex()).compareTo(Integer.valueOf(p_compareTo_1_.func_188419_a().getSortIndex()));
```

Comment:
i think you could simplify this to the following:
```Java
+ int cmp = Integer.compare(this.getPotion().getSortIndex(), that.getPotion().getSortIndex());
+ if (cmp != 0) return cmp;
```

then you won't need to do a complicated patch on the vanilla logic